### PR TITLE
Poroelastic Wave Example

### PR DESCRIPTION
**Pull request by rwalkerlewis**
This example is based off of the existing elastic model, with some modifications. It isn't quite fully finished - it needs work with unit scaling, and really the PoroelasticModel shouldn't require Vp and Vs fields as initial input, but at least this is a step toward putting things on the map.

### CODE REVIEW

File: examples/seismic/elastic/elastic_example.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def run(shape=(50, 50), spacing=(20.0, 20.0), tn=1000.0,
     info("Applying Forward")
     # Define receiver geometry (spread across x, just below surface)
     rec1, rec2, vx, vz, txx, tzz, txz, summary = solver.forward(autotune=autotune)
-
+    from IPython import embed;embed()
```
Comment (mloubout) [inline examples/seismic/elastic/elastic_example.py]: remove debug.

**Comment by mloubout** [comment]
Updates?

**Comment by rwalkerlewis** [comment]
I haven't had a chance to look at this in a little bit - let me try a revision taking into account some of the updates to the seismic code you guys put in, plus basing it around those units (same as elastic, km, ms, etc)

File: examples/seismic/model.py [hunk 2]
```diff
@@ -201,7 +272,7 @@ def demo_model(preset, **kwargs):
         # A simple circle in a 2D domain with a background velocity.
         # By default, the circle velocity is 2.5 km/s,
         # and the background veloity is 3.0 km/s.
```
Comment (georgebisbas) [inline examples/seismic/model.py]: velocity

File: examples/seismic/model.py [hunk 3]
```diff
@@ -681,3 +752,131 @@ def critical_dt(self):
         # The CFL condtion is then given by
```
Comment (georgebisbas) [inline examples/seismic/model.py]: condition

File: examples/seismic/model.py [hunk 4]
```diff
@@ -681,3 +752,131 @@ def critical_dt(self):
...
+    :param k: Formation permeability, m**2 (isotropic)
+    :param mu_f: Fluid Viscosity, Pa*s  
+    :param K_dr: Drained Bulk Modulus, Pa 
+    :param K_s: Solid Bulk Modulus, Pa
+    :param K_f: Fluid Modulus, Pa 
+    :param G: Frame Shear Modulus, Pa
+    :param T: Formation Tortuosity
+    The :class:`ModelPoroelastic` provides a symbolic data objects for the
+    creation of seismic wave propagation operators:
+    :param damp: The damping field for absorbing boundary condition
```
Comment (georgebisbas) [inline examples/seismic/model.py]: condition**s**

File: examples/seismic/model.py [hunk 5]
```diff
@@ -681,3 +752,131 @@ def critical_dt(self):
...
+            initialize_function(self.T, T, self.nbpml)
+        else:
+            self.T = Constant(name="T", value=T)
+        self._physical_parameters += ('T',)
+
+
+    @property
+    def critical_dt(self):
+        """Critical computational time step value from the CFL condition."""
+        # For a fixed time order this number goes down as the space order increases.
```
Comment (georgebisbas) [inline examples/seismic/model.py]: decreases instead of goes down?

File: examples/seismic/poroelastic/operators.py [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    # Lame Parameter of Saturated Medium
+    l_u = K_G - 2.0/3.0 * G                 # Pa
+    
+    # Create symbols for forward wavefield, source and receivers
+    vx, vy, vz = particle_velocity_fields(model, save, space_order)
+    qx, qy, qz = relative_velocity_fields(model, save, space_order)    
+    txx, tyy, tzz, _, txz, _ = stress_fields(model, save, space_order)
+    p = pressure_fields(model, save, space_order) # Different order needed?
+    
+    # Convenience terms for nightmarish poroelastodynamic FD stencils
```
Comment (georgebisbas) [inline examples/seismic/poroelastic/operators.py]: nightmarish? Maybe something different..?

File: examples/seismic/poroelastic/operators.py [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+    rho_m = T * (rho_f/phi)  # Effective fluid density / mass coupling coefficient, kg/m**3
+    _b = mu_f / k         # Fluid Mobility / resistiving damping, (Pa * s) / m**2 = kg / (m**3 * s)
```
Comment (georgebisbas) [inline examples/seismic/poroelastic/operators.py]: **resistive** damping?

**Comment by mloubout** [comment]
Is this gonna be updated and merged?

**Comment by rwalkerlewis** [comment]
> Is this gonna be updated and merged?

Yes, I need to redo the code after working out stability analysis for the problem separately. It will be best just to redo the implementation based around the current devito master,

**Pull request closed without merging**